### ISSUE TITLE
训练结果val的指标为0

### ISSUE THREAD

**Issue by katie312** [opendatalab/DocLayout-YOLO#77]
您好！我正在使用个人数据微调doclayout，其中val训练设置已设为true，但results结果里面val相关的指标都为0，其他指标是正常的，请问是要需要怎么设置呀？感谢！

**Comment by JulioZhao97** [comment]
@katie312 您好，请问可以提供具体的command，setting，以及输出吗

**Comment by katie312** [comment]
> @katie312 您好，请问可以提供具体的command，setting，以及输出吗，以便分析

感谢回复！val.txt里面是有路径的，load也正常
command：
`python train.py --data xx --model m-doclayout --epoch 500 --image-size 1600 --batch-size 6 --project public_dataset/xx --plot 1 --optimizer SGD --lr0 0.04 --pretrain /xx/ocr_project/DocLayout-YOLO/DocLayout-YOLO-D4LA-Docsynth300K_pretrained/doclayout_yolo_d4la_imgsz1600_docsynth_pretrain.pt`

setting：
```
task: detect
mode: train
model: /xx/ocr_project/DocLayout-YOLO/DocLayout-YOLO-D4LA-Docsynth300K_pretrained/doclayout_yolo_d4la_imgsz1600_docsynth_pretrain.pt
data: d4la.yaml
epochs: 500
time: null
patience: 100
batch: 6
imgsz: 1600
save: true
save_period: 10
val_period: 1
cache: false
device: '0'
workers: 4
project: public_dataset/xx
name: yolov10m-doclayout_xx_epoch500_imgsz1600_bs6_pretrain_unknown
exist_ok: false
pretrained: true
optimizer: SGD
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: null
amp: true
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.04
lrf: 0.01
momentum: 0.9
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
auto_augment: randaugment
erasing: 0.4
crop_fraction: 1.0
cfg: null
tracker: botsort.yaml
save_dir: xx/yolov10m-doclayout_d4la_epoch500_imgsz1600_bs6_pretrain_unknown

```
result：
```
	                  epoch	           train/box_om	           train/cls_om	           train/dfl_om	           train/box_oo	           train/cls_oo	           train/dfl_oo	   metrics/precision(B)	      metrics/recall(B)	       metrics/mAP50(B)	    metrics/mAP50-95(B)	             val/box_om	             val/cls_om	             val/dfl_om	             val/box_oo	             val/cls_oo	             val/dfl_oo	                 lr/pg0	                 lr/pg1	                 lr/pg2
1	                      1	                0.55384	                0.44584	                0.85357	                0.59914	                0.43085	                0.86497	                0.80388	                0.77103	                0.80823	                0.67881	                      0	                      0	                      0	                      0	                      0	                      0	               0.080014	               0.013324	               0.013324
2	                      2	                 0.5698	                0.48716	                0.85753	                0.62131	                0.47559	                0.87083	                 0.7995	                0.74304	                0.79429	                0.66427	                      0	                      0	                      0	                      0	                      0	                      0	               0.059961	               0.026605	               0.026605
3	                      3	                0.59913	                0.55971	                  0.871	                0.65589	                0.56112	                0.88598	                 0.7515	                0.73821	                0.76304	                 0.6303	                      0	                      0	                      0	                      0	                      0	                      0	               0.039855	               0.039833	               0.039833
4	                      4	                0.62179	                0.61856	                0.88125	                0.68187	                0.63269	                0.89774	                0.75022	                0.72811	                0.75163	                0.61047	                      0	                      0	                      0	                      0	                      0	                      0	               0.039762	               0.039762	               0.039762
5	                      5	                0.62195	                0.63821	                 0.8824	                 0.6846	                0.65728	                0.89943	                0.72567	                0.72908	                0.76028	                0.62417	                      0	                      0	                      0	                      0	                      0	                      0	               0.039762	               0.039762	               0.039762
6	                      6	                0.62677	                0.64169	                0.88411	                0.68981	                 0.6626	                0.90105	                0.76874	                0.71866	                0.75482	                 0.6207	                      0	                      0	                      0	                      0	                      0	                      0	               0.039683	               0.039683	               0.039683
7	                      7	                0.62832	                0.64889	                0.88397	                0.69241	                0.66758	                0.90176	                0.76259	                0.72195	                0.76615	                0.62907	                      0	                      0	                      0	                      0	                      0	                      0	               0.039604	               0.039604	               0.039604
8	                      8	                0.62522	                0.64843	                0.88424	                0.69044	                0.66557	                0.90149	                 0.8017	                0.70601	                0.76977	                0.63211	                      0	                      0	                      0	                      0	                      0	                      0	               0.039525	               0.039525	               0.039525
9	                      9	                0.62978	                0.64792	                 0.8845	                0.69481	                0.66862	                0.90288	                0.77749	                0.72241	                0.76204	                0.63186	                      0	                      0	                      0	                      0	                      0	                      0	               0.039446	               0.039446	               0.039446
10	                     10	                0.62551	                0.64506	                0.88464	                0.69044	                0.66499	                0.90296	                0.73851	                0.72396	                0.75377	                0.61708	                      0	                      0	                      0	                      0	                      0	                      0	               0.039366	               0.039366	               0.039366
11	                     11	                 0.6318	                0.64486	                0.88658	                 0.6956	                0.66691	                0.90538	                0.77448	                0.74155	                0.77658	                0.64218	                      0	                      0	                      0	                      0	                      0	                      0	               0.039287	               0.039287	               0.039287
12	                     12	                0.62555	                0.63887	                0.88528	                0.69002	                0.66281	                0.90364	                0.79388	                0.72335	                0.76767	                0.63709	                      0	                      0	                      0	                      0	                      0	                      0	               0.039208	               0.039208	               0.039208
13	                     13	                0.62509	                0.63863	                0.88221	                0.68976	                0.65803	                 0.9003	                0.77663	                0.73956	                 0.7748	                0.64032	                      0	                      0	                      0	                      0	                      0	                      0	               0.039129	               0.039129	               0.039129
14	                     14	                0.62373	                0.63544	                0.88329	                0.68699	                 0.6542	                0.90133	                0.80139	                0.69637	                0.76995	                0.63552	                      0	                      0	                      0	                      0	                      0	                      0	                0.03905	                0.03905	                0.03905
15	                     15	                0.62411	                0.64134	                0.88372	                 0.6889	                0.66175	                0.90219	                0.78315	                0.71278	                0.76723	                 0.6363	                      0	                      0	                      0	                      0	                      0	                      0	                0.03897	                0.03897	                0.03897
16	                     16	                0.62773	                0.64073	                0.88371	                0.69285	                0.66526	                0.90137	                0.78604	                0.71598	                0.77458	                0.64025	                      0	                      0	                      0	                      0	                      0	                      0	               0.038891	               0.038891	               0.038891
17	                     17	                0.62376	                0.63388	                0.88328	                0.68884	                0.66031	                0.90159	                0.77381	                0.72896	                0.77458	                 0.6409	                      0	                      0	                      0	                      0	                      0	                      0	               0.038812	               0.038812	               0.038812
18	                     18	                0.62523	                0.63418	                0.88268	                0.68951	                0.65544	                0.90057	                0.78371	                0.72363	                0.77122	                0.64032	                      0	                      0	                      0	                      0	                      0	                      0	               0.038733	               0.038733	               0.038733
19	                     19	                0.62496	                0.63741	                 0.8843	                 0.6886	                0.66146	                0.90206	                0.76751	                0.73294	                0.77082	                0.63824	                      0	                      0	                      0	                      0	                      0	                      0	               0.038654	               0.038654	               0.038654
20	                     20	                0.62372	                0.63335	                0.88275	                0.68548	                0.65157	                0.90057	                0.79699	                0.74068	                0.78777	                0.65463	                      0	                      0	                      0	                      0	                      0	                      0	               0.038574	               0.038574	               0.038574
21	                     21	                0.62088	                0.62911	                0.88139	                0.68452	                0.64898	                0.89964	                0.78485	                0.74535	                0.78756	                  0.658	                      0	                      0	                      0	                      0	                      0	                      0	               0.038495	               0.038495	               0.038495
22	                     22	                0.62334	                0.63176	                0.88312	                 0.6869	                0.65316	                0.90102	                0.77773	                0.75598	                0.78967	                0.66027	                      0	                      0	                      0	                      0	                      0	                      0	               0.038416	               0.038416	               0.038416
23	                     23	                0.62045	                0.63433	                0.88176	                0.68365	                 0.6533	                0.89984	                0.78835	                0.73025	                0.78657	                0.65458	                      0	                      0	                      0	                      0	                      0	                      0	               0.038337	               0.038337	               0.038337
24	                     24	                0.61809	                0.62701	                0.87989	                0.68296	                0.64868	                0.89779	                0.77984	                0.73746	                0.78013	                 0.6516	                      0	                      0	                      0	                      0	                      0	                      0	               0.038258	               0.038258	               0.038258
25	                     25	                0.62079	                0.62496	                0.88167	                 0.6848	                0.64349	                0.89897	                0.76871	                0.76149	                0.78459	                0.65375	                      0	                      0	                      0	                      0	                      0	                      0	               0.038178	               0.038178	               0.038178
26	                     26	                 0.6217	                0.62934	                0.88309	                0.68448	                0.64951	                 0.9008	                0.76656	                0.74455	                0.79114	                0.66051	                      0	                      0	                      0	                      0	                      0	                      0	               0.038099	               0.038099	               0.038099
27	                     27	                0.62003	                 0.6241	                0.88201	                0.68263	                0.64475	                0.89925	                0.80079	                0.74355	                0.79339	                0.66441	                      0	                      0	                      0	                      0	                      0	                      0	                0.03802	                0.03802	                0.03802
28	                     28	                0.62177	                 0.6214	                0.88107	                0.68527	                0.64207	                0.89899	                0.78747	                0.74721	                0.77916	                0.64683	                      0	                      0	                      0	                      0	                      0	                      0	               0.037941	               0.037941	               0.037941
29	                     29	                0.61868	                0.62073	                0.88128	                0.68421	                0.64116	                0.89959	                0.81711	                0.73785	                0.78909	                 0.6626	                      0	                      0	                      0	                      0	                      0	                      0	               0.037862	               0.037862	               0.037862
30	                     30	                 0.6176	                0.61936	                 0.8816	                0.68103	                0.63654	                0.89905	                0.78949	                0.73805	                0.78837	                0.65963	                      0	                      0	                      0	                      0	                      0	                      0	               0.037782	               0.037782	               0.037782
31	                     31	                0.61527	                0.62511	                0.87956	                0.67961	                0.64324	                0.89735	                 0.7797	                0.75541	                0.79356	                0.66647	                      0	                      0	                      0	                      0	                      0	                      0	               0.037703	               0.037703	               0.037703
32	                     32	                0.61653	                0.61671	                0.88122	                0.68072	                 0.6345	                0.89922	                0.80593	                0.74461	                0.79651	                0.66721	                      0	                      0	                      0	                      0	                      0	                      0	               0.037624	               0.037624	               0.037624
33	                     33	                0.61663	                0.61561	                0.87914	                0.67874	                0.63292	                0.89596	                0.81177	                  0.756	                0.80106	                0.67078	                      0	                      0	                      0	                      0	                      0	                      0	               0.037545	               0.037545	               0.037545
34	                     34	                0.61736	                0.61884	                0.88153	                 0.6788	                0.63553	                0.89845	                0.79966	                0.75144	                0.79895	                0.67053	                      0	                      0	                      0	                      0	                      0	                      0	               0.037466	               0.037466	               0.037466
35	                     35	                0.61323	                0.61184	                0.87881	                0.67467	                0.62897	                0.89592	                0.79058	                0.75446	                0.80178	                 0.6713	                      0	                      0	                      0	                      0	                      0	                      0	               0.037386	               0.037386	               0.037386
36	                     36	                0.61822	                0.61247	                0.87954	                0.68034	                0.62856	                0.89615	                0.78726	                0.75292	                0.79715	                0.66913	                      0	                      0	                      0	                      0	                      0	                      0	               0.037307	               0.037307	               0.037307
37	                     37	                0.61615	                 0.6108	                0.88092	                0.67713	                0.62977	                0.89769	                0.79331	                0.75697	                0.79832	                  0.673	                      0	                      0	                      0	                      0	                      0	                      0	               0.037228	               0.037228	               0.037228
38	                     38	                0.61814	                0.61406	                0.88062	                0.68074	                0.63043	                0.89753	                0.80922	                0.73057	                0.79255	                0.66464	                      0	                      0	                      0	                      0	                      0	                      0	               0.037149	               0.037149	               0.037149
39	                     39	                0.61674	                0.61034	                0.88127	                0.67758	                0.62763	                0.89822	                0.79378	                0.75753	                0.79707	                0.66913	                      0	                      0	                      0	                      0	                      0	                      0	                0.03707	                0.03707	                0.03707
40	                     40	                0.61192	                0.60709	                0.87947	                0.67322	                0.62288	                0.89647	                0.78632	                0.74221	                0.78426	                 0.6573	                      0	                      0	                      0	                      0	                      0	                      0	                0.03699	                0.03699	                0.03699
41	                     41	                0.61348	                0.60594	                0.87646	                0.67489	                  0.624	                0.89311	                 0.8051	                0.74756	                 0.8005	                0.67108	                      0	                      0	                      0	                      0	                      0	                      0	               0.036911	               0.036911	               0.036911
42	                     42	                0.61489	                0.60572	                0.87899	                0.67536	                0.62179	                0.89589	                  0.788	                0.76194	                0.80421	                0.67609	                      0	                      0	                      0	                      0	                      0	                      0	               0.036832	               0.036832	               0.036832
43	                     43	                0.61165	                 0.6054	                  0.878	                0.67298	                0.61728	                0.89425	                0.81119	                0.75327	                0.80364	                0.67463	                      0	                      0	                      0	                      0	                      0	                      0	               0.036753	               0.036753	               0.036753
44	                     44	                 0.6152	                0.60884	                0.87977	                0.67738	                0.62491	                0.89683	                0.83143	                0.73809	                 0.8089	                0.68125	                      0	                      0	                      0	                      0	                      0	                      0	               0.036674	               0.036674	               0.036674
45	                     45	                0.61355	                0.60392	                0.87856	                0.67587	                0.61661	                0.89541	                0.81575	                0.75259	                0.79776	                 0.6739	                      0	                      0	                      0	                      0	                      0	                      0	               0.036594	               0.036594	               0.036594
46	                     46	                0.61064	                0.59909	                0.87536	                0.67329	                0.61185	                0.89267	                0.79002	                0.76219	                0.80053	                0.67456	                      0	                      0	                      0	                      0	                      0	                      0	               0.036515	               0.036515	               0.036515
47	                     47	                 0.6101	                0.60155	                0.87703	                0.67071	                0.61544	                0.89384	                0.80219	                0.75956	                0.80366	                0.67579	                      0	                      0	                      0	                      0	                      0	                      0	               0.036436	               0.036436	               0.036436
48	                     48	                0.61031	                0.60246	                0.87584	                0.67172	                0.61624	                0.89152	                0.81403	                0.75536	                0.80479	                0.67721	                      0	                      0	                      0	                      0	                      0	                      0	               0.036357	               0.036357	               0.036357
49	                     49	                0.61254	                0.60542	                 0.8763	                0.67208	                0.62029	                0.89218	                0.79522	                0.76738	                  0.809	                0.68242	                      0	                      0	                      0	                      0	                      0	                      0	               0.036278	               0.036278	               0.036278
50	                     50	                0.61207	                0.59895	                0.87547	                0.67424	                0.61295	                0.89131	                0.82139	                 0.7618	                0.80789	                0.68228	                      0	                      0	                      0	                      0	                      0	                      0	               0.036198	               0.036198	               0.036198
51	                     51	                0.61091	                0.59826	                0.87782	                0.67259	                0.61113	                0.89426	                0.81387	                0.76127	                0.80492	                0.67667	                      0	                      0	                      0	                      0	                      0	                      0	               0.036119	               0.036119	               0.036119
52	                     52	                0.61266	                0.60187	                0.87913	                  0.675	                0.61774	                0.89607	                0.81121	                0.76829	                0.80828	                0.68065	                      0	                      0	                      0	                      0	                      0	                      0	                0.03604	                0.03604	                0.03604
53	                     53	                 0.6181	                0.60986	                0.87957	                 0.6794	                0.62457	                0.89553	                0.83898	                0.75129	                0.81374	                0.68547	                      0	                      0	                      0	                      0	                      0	                      0	               0.035961	               0.035961	               0.035961
54	                     54	                0.61004	                 0.5981	                0.87762	                0.67125	                0.61435	                0.89352	                 0.7998	                0.77635	                0.81325	                0.68529	                      0	                      0	                      0	                      0	                      0	                      0	               0.035882	               0.035882	               0.035882
```

**Comment by JulioZhao97** [comment]
您好，这里是经过修改的不去evalute loss，只evaluate指标，如果您想要evaluate loss需要打开以下注释掉的地方：
https://github.com/opendatalab/DocLayout-YOLO/blob/afeb639f981e47ae1961fb0c1ba293250b49b954/doclayout_yolo/engine/validator.py#L172-L208

**Comment by katie312** [comment]
感谢！我试试
